### PR TITLE
Updating Section to fit NeMo Framework User Guide

### PR DESCRIPTION
**Pull request by aschilling-nv**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

Updating Section to fit NeMo Framework User Guide

# Changelog 
- Please update the [CHANGELOG.md](/CHANGELOG.md) under next version with high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation? Make sure to also update the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/index.html) which contains the tutorials

# Checklist when contributing a new algorithm
- [ ] Does the trainer resume and restore model state all states?
- [ ] Does the trainer support all parallelism techniques(PP, TP, DP)?
- [ ] Does the trainer support `max_steps=-1` and `validation`?
- [ ] Does the trainer only call APIs defined in [alignable_interface.py](/nemo_aligner/models/alignable_interface.py)?
- [ ] Does the trainer have proper logging?

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by gshennvm — CHANGES_REQUESTED** [review]

File: docs/user-guide/index.rst
Comment (gshennvm) [inline docs/user-guide/index.rst:24]: do we not need to update the SFT.rst file?

Comment (aschilling-nv) [reply docs/user-guide/index.rst:24]: The SFT.rst file already had anchor links in the appropriate places.

https://github.com/NVIDIA/NeMo-Aligner/blob/main/docs/user-guide/SFT.rst?plain=1#L3
https://github.com/NVIDIA/NeMo-Aligner/blob/main/docs/user-guide/SFT.rst?plain=1#L49

File: docs/user-guide/index.rst
Comment (gshennvm) [inline docs/user-guide/index.rst:8]: sorry one last thing -- can this be updated to <model-aligner-sft> once this is updated we can merge

Comment (thomasdhc) [reply docs/user-guide/index.rst:8]: @gshennvm What's the update required here?

Comment (thomasdhc) [reply docs/user-guide/index.rst:8]: https://github.com/aschilling-nv/NeMo-Aligner/blob/main/docs/user-guide/SFT.rst?plain=1#L49

Comment (gshennvm) [reply docs/user-guide/index.rst:8]: right, can we change that to model-aligner-sft?

Comment (thomasdhc) [reply docs/user-guide/index.rst:8]: Yeah, @aschilling-nv I don't have permission to push to the fork. Could you update when you get a chance? Thanks!

**Comment by thomasdhc** [comment]
@gshennvm Could you do a final review?

**Review by gshennvm — APPROVED** [review]

**Comment by gshennvm** [comment]
> @gshennvm Could you do a final review?

good to merge once DCO is resolved

**Pull request merged**
